opus/application/apps/metadata/views.py

```diff
@@ -762,15 +781,21 @@ def get_fields_info(fmt, request, api_code, slug=None, collapse=False):
             else:
                 entry['old_slug'] = f.old_slug
             entry['slug'] = entry['field_id'] # Backwards compatibility
-            return_obj[collapsed_slug] = entry
+            entry['referred'] = True if f.referred_slug else False
+            return_obj.get(cat)[collapsed_slug] = entry
 
+        # Organize return_obj before returning
+        # Sort categories by table_order
         return_obj = OrderedDict(sorted(return_obj.items(),
-                                        key=lambda x: (x[1]['table_order'],
-                                                       x[1]['disp_order'])))
-        # Hide internal sort order info from the end user
-        for key, val in return_obj.items():
-            del val['table_order']
-            del val['disp_order']
+                                    key=lambda x: x[1]['table_order']))
+        for cat, cat_data in return_obj.items():
+            del cat_data['table_order']
+        # Sort slugs of each category by disp_order
+        return_obj[cat] = OrderedDict(sorted(return_obj[cat].items(),
```

Comment:
I'm confused. Shouldn't this be indented so it's part of the loop? In which case you don't need return_obj[cat] since that's just cat_data. The way it is now you're just sorting the last category. When this is fixed you can combine it with the loop below like it was before.